core/modules/review/store/actions.ts

```diff
@@ -1,93 +1,34 @@
-import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
-import { adjustMultistoreApiUrl } from '@vue-storefront/core/lib/multistore'
 import RootState from '@vue-storefront/core/types/RootState'
 import ReviewState from '../types/ReviewState'
 import * as types from './mutation-types'
 import i18n from '@vue-storefront/i18n'
-import rootStore from '@vue-storefront/core/store'
 import Review from '@vue-storefront/core/modules/review/types/Review'
-import { ReviewRequest } from '@vue-storefront/core/modules/review/types/ReviewRequest'
-import { Logger } from '@vue-storefront/core/lib/logger'
-import config from 'config'
-import { processURLAddress } from '@vue-storefront/core/helpers'
+import { createQueryLoadingQuery, notifications } from '@vue-storefront/core/modules/review/helpers'
+import { ReviewsService } from '@vue-storefront/core/data-resolver'
 
 const actions: ActionTree<ReviewState, RootState> = {
-  /**
-   * Retrieve reviews
-   *
-   * @param context
-   * @param {any} query
-   * @param {any} start
-   * @param {any} size
-   * @param {any} entityType
-   * @param {any} sort
-   * @param {any} excludeFields
-   * @param {any} includeFields
-   * @returns {Promise<T> & Promise<any>}
-   */
-  list (context, {productId, approved = true, start = 0, size = 50, entityType = 'review', sort = '', excludeFields = null, includeFields = null}) {
-    let query = new SearchQuery()
+  async list (context, {productId, approved = true, start = 0, size = 50, entityType = 'review', sort = '', excludeFields = null, includeFields = null}) {
+    const query = createQueryLoadingQuery({ productId, approved })
 
-    if (productId) {
-      query = query.applyFilter({key: 'product_id', value: {'eq': productId}})
-    }
-
-    if (approved) {
-      query = query.applyFilter({key: 'review_status', value: {'eq': 1}})
-    }
-
-    quickSearchByQuery({ query, start, size, entityType, sort, excludeFields, includeFields }).then((resp) => {
-      context.commit(types.REVIEW_UPD_REVIEWS, resp)
-    }).catch(err => {
-      Logger.error(err, 'review')()
-    })
+    const reviewResponse = await quickSearchByQuery({ query, start, size, entityType, sort, excludeFields, includeFields })
+    context.commit(types.REVIEW_UPD_REVIEWS, reviewResponse)
   },
-
-  /**
-   * Add new review
-   *
-   * @param context
-   * @param {Review} reviewData
-   * @returns {Promise<void>}
-   */
-  async add (context, reviewData: Review) {
-    const review: ReviewRequest = {review: reviewData}
-
+  async add ({ dispatch }, review: Review) {
     EventBus.$emit('notification-progress-start', i18n.t('Adding a review ...'))
 
-    let url = config.reviews.create_endpoint
+    const reviewResponse = await ReviewsService.createReview(review)
+    EventBus.$emit('notification-progress-stop')
 
-    if (config.storeViews.multistore) {
-      url = adjustMultistoreApiUrl(url)
+    if (reviewResponse.code === 200) {
```

Comment:
would be better if `ReviewsService.createReview` would return only `true/added review object` and `false/throw an error` when something is wrong, then we don't need to carry and play with the response codes